astroplan/scheduling.py

```diff
@@ -90,6 +92,81 @@ def from_exposures(cls, target, priority, time_per_exposure,
         return ob
 
 
+class Scorer(object):
+    """
+    Returns scores and score arrays from the evaluation of constraints on
+    observing blocks
+    """
+    def __init__(self, blocks, observer, schedule):
+        """
+
+        Parameters
+        ----------
+        blocks : list of `~astroplan.scheduling.ObservingBlock` objects
+            list of blocks that need to be scored
+        observer : `~astroplan.Observer`
+            the observer
+        schedule : `~astroplan.scheduling.Schedule`
+            The schedule inside which the blocks should fit
+        """
+        self.blocks = blocks
+        self.observer = observer
+        self.schedule = schedule
+        # the ranges below should change if there are new constraints made
+        self.constraint_list = [eval(name) for name in
+                                astroplan.constraints.__all__[:-6]
+                                if 'Constraint' in name]
+        self.scheduling_constraints = [eval(name) for name in
+                                       astroplan.constraints.__all__[-5:]
+                                       if 'Constraint' in name]
```

Comment:
The use of eval here could cause issues (pointed out by @StuartLittlefair). @bmorris3 is it possible to make these lists part of the `Constraints`? That way it could just call the list from `Constraints`.